src/storage/src/hummock/local_version.rs

```diff
@@ -367,36 +451,45 @@ impl LocalVersion {
         &mut self,
         max_committed_epoch: HummockEpoch,
     ) -> Vec<(Vec<HummockEpoch>, Vec<LocalSstableInfo>)> {
-        match self.sync_uncommitted_data.iter().position(|(epoch, data)| {
-            let min_epoch = *epoch.last().expect("epoch list should not be empty");
-            let max_epoch = *epoch.first().expect("epoch list should not be empty");
-            assert!(
-                max_epoch <= max_committed_epoch || min_epoch > max_committed_epoch,
-                "new_max_committed_epoch {} lays within max_epoch {} and min_epoch {} of data {:?}",
-                max_committed_epoch,
-                max_epoch,
-                min_epoch,
-                data,
-            );
-            max_epoch <= max_committed_epoch
+        match self.sync_uncommitted_data
+            .iter()
+            .rev() // Take rev so that newer epochs come first
+            .find(|(sync_epoch, data)| {
+            if data.epochs.is_empty() {
```

Comment:
When will data.epochs.is_empty() == true?

Reply:
When no data is written between prev max_sync_epoch and current sync epoch, `data.epochs` is empty. We store this empty task so that in the future PR that decouples `seal_epoch` and `await_sync_epoch`, we can collect the empty epoch in `await_sync_epoch`.